netlogo-gui/src/test/ide/JumpToDeclarationTests.scala

```diff
@@ -0,0 +1,97 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.ide
+
+import org.nlogo.core.{Token, TokenType}
+import org.scalatest.FunSuite
+
+class JumpToDeclarationTests extends FunSuite {
+
+  val source1 =
+    """
+    |me-own[
+    |  variable
+    |]
+    |
+    |you-own [
+    |  variable
+    |]
+    |
+    |globals [scared]
+    |to foo
+    |  scared 1
+    |  let scared
+    |  set scared 0
+    |end
+    |
+    |to new [scared]
+    |  set variable 7
+    |  set scared 5
+    |end
+    """.stripMargin
+
+  val source2 =
+    """
+      |to foo
+      |[
+      |  let scared
+      |  set scared 0
+      |]
+      |set scared 5
+      |end
+    """.stripMargin
+
+  def createToken(text: String, tokenType: TokenType, startPos: Int) = new Token(text, tokenType, null)(startPos, 0, null)
+
+  def testTokenPosition(tokenToFindDeclaration: Token, startIndexOfDeclaration: Int): Unit = {
+      val tokenOption = JumpToDeclaration.getDeclaration(tokenToFindDeclaration, source1)
+      assertResult(true)(tokenOption.isDefined)
```

Comment:
This is OK as is, but you could also write it as `assert(tokenOption.isDefined)`.